src/local-run.ts

```diff
@@ -1,51 +1,131 @@
+    "run",
+    "-q",
+    "--config=deno.jsonc",
+    "--allow-read",
+  ];
+
+  const allowedDomains = manifest.outgoing_domains ?? [];
+
+  // If using a dev instance, allow making API calls to that domain
+  // and ignore SSL errors to it
+  if (devDomain !== "") {
+    command.push("--unsafely-ignore-certificate-errors=" + devDomain);
+    allowedDomains.push(devDomain);
+  } else {
+    allowedDomains.push("slack.com");
+  }
+  // Add deno.land to allow uncached remote deps
+  allowedDomains.push("deno.land");
+
+  command.push("--allow-net=" + allowedDomains.join(","));
+  command.push(findRelativeFile(mainModule, "local-run-function.ts"));
+
+  return command;
+};
 
 /**
- * @description Runs an application function locally by dispatching a payload to it after loading it.
+ * @description Runs an application locally by calling `deno run` with appropriate flags.
  */
-export const runLocally = async function (
+export const runWithOutgoingDomains = async function (
   create: typeof createManifest,
-  parse: typeof ParsePayload,
-  readStdin: typeof readAll,
-  dispatch: typeof DispatchPayload,
+  devDomain: string,
+  // deno-lint-ignore no-explicit-any
+  log: (...any: any) => void,
 ): Promise<void> {
   const workingDirectory = Deno.cwd();
   const manifest = await create({
     manifestOnly: true,
-    log: () => {},
+    log: log,
     workingDirectory,
   });
+
   if (!manifest.functions) {
     throw new Error(
       `No function definitions were found in the manifest! manifest.functions: ${manifest.functions}`,
     );
   }
-  const payload = await parse(readStdin);
-
-  // Finds the corresponding function in the manifest definition, and then uses
-  // the `source_file` property to determine the function module file location
-  const resp = await dispatch(payload, (functionCallbackId) => {
-    const functionDefn = manifest.functions[functionCallbackId];
-    if (!functionDefn) {
-      throw new Error(
-        `No function definition for function callback id ${functionCallbackId} was found in the manifest! manifest.functions: ${manifest.functions}`,
-      );
-    }
-
-    const functionFile =
-      `file://${workingDirectory}/${functionDefn.source_file}`;
-
-    return functionFile;
+
+  let denoExecutablePath = "deno";
+  try {
+    denoExecutablePath = Deno.execPath();
+  } catch (e) {
+    log("Error determining deno executable path: ", e);
+  }
+
+  const command = getCommandline(
+    Deno.mainModule,
+    denoExecutablePath,
+    manifest,
+    devDomain,
+  );
+
+  log("running command: ", ...command);
+
+  const p = Deno.run({
+    cmd: command,
   });
 
-  // The CLI expects a JSON payload to be output to stdout
-  // This is formalized in the `run` hook of the CLI/SDK Tech Spec:
-  // https://corp.quip.com/0gDvAsqoaaYE/Proposal-CLI-SDK-Interface#temp:C:fOC1991c5aec8994d0db01d26260
-  console.log(JSON.stringify(resp || {}));
+  const status = await p.status();
+  if (!status.success) {
+    Deno.exit(status.code);
```

Comment:
Makes sense to mimic the exit code for the housing process to be the same as the sub-process. For the record, the invoked deno process' status code is not used at all on the CLI side: https://github.com/slackapi/slack-cli/blob/main/pkg/cloud/run.go#L299-L303

Reply:
Ohhhh rightttttt my bad. Good call out.